Error when compiling on Arch Linux

I get following error when compiling on Arch Linux

`/home/lethe/lethean/src/cryptonote_basic/miner.cpp: In member function ‘bool cryptonote::miner::background_worker_thread()’:
/home/lethe/lethean/src/cryptonote_basic/miner.cpp:671:24: error: cannot convert ‘boost::logic::tribool’ to ‘bool’ in assignment
         on_ac_power = !battery_powered;`

--- a/src/cryptonote_basic/miner.cpp
+++ b/src/cryptonote_basic/miner.cpp
@@ -668,7 +668,7 @@ namespace cryptonote
       }
       else
       {
-        on_ac_power = !battery_powered;
+        on_ac_power = !(bool)battery_powered;
       }
 
       if( m_is_background_mining_started )

